projects/optic/src/commands/diff/diff-all.ts

```diff
@@ -0,0 +1,365 @@
+
+        if (shouldUploadResults) {
+          // TODO upload results
+        }
+      }
+      results.push({
+        specResults,
+        checks,
+        from: candidate.from,
+        to: candidate.to,
+      });
+
+      if (
+        specResults.changes.length > 0 ||
+        (!options.check && specResults.results.length > 0)
+      ) {
+        const meta = {
+          createdAt: new Date(),
+          command: ['optic', ...process.argv.slice(2)].join(' '),
+        };
+
+        const compressedData = compressData(
+          fromParseResults,
+          toParseResults,
+          specResults,
+          meta
+        );
+        const anonymousId = await getAnonId();
+        trackEvent('optic.diff_all.view_web', anonymousId, {
+          compressedDataLength: compressedData.length,
+        });
+        await open(`${config.client.getWebBase()}/cli/diff#${compressedData}`, {
+          wait: false,
+        });
+      }
+    } else if (
+      !toParseResults.isEmptySpec &&
+      typeof toParseResults.jsonLike[OPTIC_URL_KEY] !== 'string'
+    ) {
+      warnings.missingOpticUrl.push({
+        path: candidate.to!,
+      });
+      continue;
+    }
+  }
+  return {
+    warnings,
+    results,
+  };
+}
+
+type Results = (Awaited<ReturnType<typeof compute>> & {
+  from?: string;
+  to?: string;
+})[];
+
+type Warnings = {
+  missingOpticUrl: {
+    path: string;
+  }[];
+  unparseableFromSpec: {
+    path: string;
+    error: unknown;
+  }[];
+  unparseableToSpec: { path: string; error: unknown }[];
+};
+
+const getDiffAllAction =
+  (config: OpticCliConfig) => async (options: DiffAllActionOptions) => {
+    if (config.vcs?.type !== VCS.Git) {
+      logger.error(
+        `Error: optic diff-all must be called from a git repository.`
+      );
+      process.exitCode = 1;
+      return;
+    }
+
+    let compareToCandidates;
+    let compareFromCandidates;
+
+    try {
+      compareToCandidates = await findOpenApiSpecsCandidates(options.compareTo);
+    } catch (e) {
+      logger.error(
+        `Error reading files from git history for --compare-to ${options.compareTo}`
+      );
+      logger.error(e);
+      process.exitCode = 1;
+      return;
+    }
+
+    try {
+      compareFromCandidates = await findOpenApiSpecsCandidates(
+        options.compareFrom
+      );
+    } catch (e) {
+      logger.error(
+        `Error reading files from git history for --compare-from ${options.compareFrom}`
+      );
+      logger.error(e);
+      process.exitCode = 1;
+      return;
+    }
+
+    const candidatesMap = matchCandidates(
+      {
+        ref: options.compareFrom,
+        paths: compareFromCandidates,
+      },
+      {
+        ref: options.compareTo,
+        paths: compareToCandidates,
+      }
+    );
+
+    const { warnings, results } = await computeAll(
+      candidatesMap,
+      config,
+      options
+    );
+
+    if (warnings.missingOpticUrl.length > 0) {
```

Comment:
I think we landed process.exitCode !== 1 in this case, but happy to change either way